scilpy/reconst/multi_processes.py

```diff
@@ -0,0 +1,420 @@
+                 nbr_processes=None):
+    """Computes maps from given SH coefficients and peaks
+
+    Parameters
+    ----------
+    shm_coeff : np.ndarray
+        Spherical harmonic coefficients
+    peak_dirs : np.ndarray
+        Peak directions
+    peak_values : np.ndarray
+        Peak values
+    peak_indices : np.ndarray
+        Peak indices
+    sphere : Sphere
+        The Sphere providing discrete directions for evaluation.
+    mask : np.ndarray, optional
+        If `mask` is provided, only the data inside the mask will be
+        used for computations.
+    gfa_thr : float, optional
+        Voxels with gfa less than `gfa_thr` are skipped for all metrics, except
+        `rgb_map`.
+        Default: 0
+    sh_basis_type : str, optional
+        Type of spherical harmonic basis used for `shm_coeff`. Either
+        `descoteaux07` or `tournier07`.
+        Default: `descoteaux07`
+    nbr_processes: int, optional
+        The number of subprocesses to use.
+        Default: multiprocessing.cpu_count()
+
+    Returns
+    -------
+    tuple of np.ndarray
+        nufo_map, afd_map, afd_sum, rgb_map, gfa, qa
+    """
+    sh_order = order_from_ncoef(shm_coeff.shape[-1])
+    B, _ = sh_to_sf_matrix(sphere, sh_order, sh_basis_type)
+
+    data_shape = shm_coeff.shape
+    if mask is not None:
+        shm_coeff = applymask(shm_coeff, mask)
+        peak_dirs = applymask(peak_dirs, mask)
+        peak_values = applymask(peak_values, mask)
+        peak_indices = applymask(peak_indices, mask)
+
+    nbr_processes = multiprocessing.cpu_count() if nbr_processes is None \
+        or nbr_processes < 0 else nbr_processes
+
+    npeaks = peak_values.shape[3]
+    shm_coeff = shm_coeff.ravel().reshape(np.prod(data_shape[0:3]), data_shape[3])
+    peak_dirs = peak_dirs.ravel().reshape((np.prod(data_shape[0:3]), npeaks, 3))
+    peak_values = peak_values.ravel().reshape(np.prod(data_shape[0:3]), npeaks)
+    peak_indices = peak_indices.ravel().reshape(np.prod(data_shape[0:3]), npeaks)
+    shm_coeff_chunks = np.array_split(shm_coeff, nbr_processes)
+    peak_dirs_chunks = np.array_split(peak_dirs, nbr_processes)
+    peak_values_chunks = np.array_split(peak_values, nbr_processes)
+    peak_indices_chunks = np.array_split(peak_indices, nbr_processes)
+    chunk_len = np.cumsum([0] + [len(c) for c in shm_coeff_chunks])
+
+    pool = multiprocessing.Pool(nbr_processes)
+    results = pool.map(maps_from_sh_parallel,
+                       zip(shm_coeff_chunks,
+                           peak_dirs_chunks,
+                           peak_values_chunks,
+                           peak_indices_chunks,
+                           itertools.repeat(B),
+                           itertools.repeat(sphere),
+                           itertools.repeat(gfa_thr),
+                           np.arange(len(shm_coeff_chunks))))
+    pool.close()
+    pool.join()
+
+    nufo_map_array = np.zeros((np.prod(data_shape[0:3])))
```

Comment:
Like the previous comments, since this is the same in the 3 functions, maybe a re-assembling and reshaping function could be useful. 

input data_shape (4D) and a list of list of 3D or 4D array and return a list array (in the same order).
These two comments are optional, only do them if someone else than me confirms.